### PR TITLE
Draw 6 players around the table

### PR DESCRIPTION
**Pull request by rictic**
Control one of them at random with the arrow keys.

### CODE REVIEW

**Review by mohammadshamma — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -10,15 +10,59 @@ use wasm_bindgen::JsCast;
 #[global_allocator]
 static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
 
-#[wasm_bindgen]
+#[derive(Clone, Copy, PartialEq, Eq)]
```
Comment (cdata) [inline src/lib.rs:13]: `derive` 😲

File: src/lib.rs [hunk 2]
```diff
@@ -10,15 +10,59 @@ use wasm_bindgen::JsCast;
...
 #[derive(Clone)]
-pub struct Game {
+pub struct Player {
+    color: Color,
     x: f64,
     y: f64,
+}
+
+#[wasm_bindgen]
+#[derive(Clone)]
```
Comment (cdata) [inline src/lib.rs:58]: porque?

File: src/lib.rs [hunk 3]
```diff
@@ -179,12 +248,43 @@ pub fn make_game() -> MakeGameResult {
...
                 context,
                 width,
                 height,
+                local_player_color: Some(Color::random()),
+                players: vec![
+                    Player {
+                        color: Color::Red,
+                        x: 275.0 + (100.0 * (1.0 / 6.0 * 2.0 * f64::consts::PI).sin()),
+                        y: 275.0 + (100.0 * (1.0 / 6.0 * 2.0 * f64::consts::PI).cos()),
+                    },
```
Comment (cdata) [inline src/lib.rs]: Could these players have been instantiated in a loop somehow?

**Review by cdata — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (rictic) [reply src/lib.rs:13]: Derive is _great_, it implements what I'd call "the most obvious implementation" of a trait for your type. So unlike Java where (unless you use compiler plugins I think) you have to describe how `==` should work by hand, or JavaScript where you can have any kind of equality you want as long as it's referential equality, in Rust you get the best of both worlds. You get to define basically everything about how your type should work, but if you just want the obvious implementation, you can usually just `derive` it.

File: src/lib.rs (hunk 2 above)
Comment (rictic) [reply src/lib.rs:58]: We clone our Game on line 234 to make the borrow checker happy. `.clone()` makes a totally separate deep copy of a value: https://doc.rust-lang.org/std/clone/trait.Clone.html

File: src/lib.rs (hunk 3 above)
Comment (rictic) [reply src/lib.rs]: +1, done

**Pull request merged**